src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_ObjectCreationExpression.cs

```diff
@@ -116,14 +117,16 @@ public override BoundNode VisitWithExpression(BoundWithExpression withExpr)
             // we want to lower it to a call to the receiver's `Clone` method, then
             // set the given record properties. i.e.
             //
-            //      var tmp = receiver.Clone();
+            //      var tmp = (ReceiverType)receiver.Clone();
             //      tmp.P1 = e1;
             //      tmp.P2 = e2;
             //      tmp
 
-            var cloneCall = _factory.Call(
-                VisitExpression(withExpr.Receiver),
-                withExpr.CloneMethod);
+            var cloneCall = _factory.Convert(
```

Comment:
>Convert [](start = 37, length = 7)

Did we check that the conversion exist? Are user-defined conversions going to be handled here? I think we should probably include conversion information into IOperation node and apply conversion in CFG rewrite when necessary. We don't have to do all this for the preview, but should capture things to follow up on.